### ISSUE TITLE
Pixmap created from CMYK JPEG delivers RGB format

### ISSUE THREAD

**Issue by JorjMcKie** [pymupdf/PyMuPDF#3058]
### Discussed in https://github.com/pymupdf/PyMuPDF/discussions/3057

<div type='discussions-op-text'>

<sup>Originally posted by **dothinking** January 18, 2024</sup>
With upgrading `PyMuPDF` from 1.23.8 to 1.23.15, fitz.Pixmap() doesn't work as expected. Do I miss something, or if it's an upstream issue.

[sample.pdf](https://github.com/pymupdf/PyMuPDF/files/13974803/sample.pdf)

## PyMuPdf 1.23.8

```python
>>> import fitz
>>> fitz.version
('1.23.8', '1.23.7', '20231219000001')
>>> doc = fitz.Document('sample.pdf')
>>> doc[0].get_images(full=True)
[(17, 0, 1731, 1331, 8, 'DeviceCMYK', '', 'Im0', 'DCTDecode', 0)]
>>> pix = fitz.Pixmap(doc, 17)
>>> pix.colorspace
Colorspace(CS_CMYK) - DeviceCMYK
>>> pix = fitz.Pixmap(fitz.csRGB, pix)
>>> pix.save('sample-1.png')                      <- perfect
```

## PyMuPdf 1.23.15

```python
>>> import fitz
>>> fitz.version
('1.23.15', '1.23.9', '20240116000001')
>>> doc = fitz.Document('sample.pdf')
>>> doc[0].get_images(full=True)
[(17, 0, 1731, 1331, 8, 'DeviceCMYK', '', 'Im0', 'DCTDecode', 0)]    <- color space = DeviceCMYK
>>> pix = fitz.Pixmap(doc, 17)
>>> pix.colorspace
Colorspace(CS_RGB) - DeviceRGB                     <- color space seems changed to DeviceRGB
>>> pix = fitz.Pixmap(fitz.csRGB, pix)
>>> pix.save('sample-2.png')                   <- the png color is different from the source
```</div>

**Comment by julian-smith-artifex-com** [comment]
Thanks for the report and for including the reproducer.

I have a fix in review, and will hopefully make a new release with the fix later today.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.16.
